selfdrive/car/mazda/carstate.py

```diff
@@ -0,0 +1,217 @@
+    super().__init__(CP)
+
+    self.steer_lkas = STEER_LKAS()
+    self.cruise_speed = 0
+    self.acc_active_last = False
+    self.speed_kph = 0
+    self.lkas_speed_lock = False
+    self.low_speed_lockout = True
+    self.low_speed_lockout_last = True
+    self.acc_press_update = False
+
+  def update(self, cp, cp_cam):
+
+    ret = car.CarState.new_message()
+    ret.wheelSpeeds.fl = cp.vl["WHEEL_SPEEDS"]['FL'] * CV.KPH_TO_MS
+    ret.wheelSpeeds.fr = cp.vl["WHEEL_SPEEDS"]['FR'] * CV.KPH_TO_MS
+    ret.wheelSpeeds.rl = cp.vl["WHEEL_SPEEDS"]['RL'] * CV.KPH_TO_MS
+    ret.wheelSpeeds.rr = cp.vl["WHEEL_SPEEDS"]['RR'] * CV.KPH_TO_MS
+    ret.vEgoRaw = (ret.wheelSpeeds.fl + ret.wheelSpeeds.fr + ret.wheelSpeeds.rl + ret.wheelSpeeds.rr) / 4.
+    ret.vEgo, ret.aEgo = self.update_speed_kf(ret.vEgoRaw)
+    ret.standstill = ret.vEgoRaw < 0.01
+
+    self.speed_kph =  ret.vEgoRaw  // CV.KPH_TO_MS
+
+    gear = int(cp.vl["GEAR"]['GEAR']) >> 1
+
+    if gear >= 1 & gear <= 5:
+      ret.gearShifter = GearShifter.drive
+    elif gear == 0:
+      ret.gearShifter = GearShifter.park
+    elif gear == 6:
+      ret.gearShifter = GearShifter.reverse
+    elif gear == 7:
+      ret.gearShifter = GearShifter.neutral
+    else:
+      ret.gearShifter = GearShifter.unknown
+
+    ret.leftBlinker = cp.vl["BLINK_INFO"]['LEFT_BLINK'] == 1
+    ret.rightBlinker = cp.vl["BLINK_INFO"]['RIGHT_BLINK'] == 1
+
+    ret.steeringAngle =  cp.vl["STEER"]['STEER_ANGLE']
+    ret.steeringTorque = cp.vl["STEER_TORQUE"]['STEER_TORQUE_SENSOR']
+    ret.steeringPressed = abs(ret.steeringTorque) > LKAS_LIMITS.STEER_THRESHOLD
+
+    self.steer_torque_motor = cp.vl["STEER_TORQUE"]['STEER_TORQUE_MOTOR']
+    self.angle_steers_rate = cp.vl["STEER_RATE"]['STEER_ANGLE_RATE']
+
+    ret.brakePressed = cp.vl["PEDALS"]['BRAKE_ON'] == 1
+    ret.brake = cp.vl["BRAKE"]['BRAKE_PRESSURE']
+    ret.brakeLights = ret.brakePressed
+
+    ret.seatbeltUnlatched = cp.vl["SEATBELT"]['DRIVER_SEATBELT'] == 0
+    ret.doorOpen = any([cp.vl["DOORS"]['FL'], cp.vl["DOORS"]['FR'],
+                        cp.vl["DOORS"]['BL'], cp.vl["DOORS"]['BR']])
+
+    ret.gas = cp.vl["ENGINE_DATA"]['PEDAL_GAS']
+    ret.gasPressed = ret.gas > 1e-3
+
+    # No steer if block signal is on
+    self.steer_lkas.block = cp.vl["STEER_RATE"]['LKAS_BLOCK']
+    # track driver torque, on if torque is not detected
+    self.steer_lkas.track = cp.vl["STEER_RATE"]['LKAS_TRACK_STATE']
+    # On if no driver torque the last 5 seconds
+    self.steer_lkas.handsoff = cp.vl["STEER_RATE"]['HANDS_OFF_5_SECONDS']
+
+    # LKAS is enabled at 50kph going up and disabled at 45kph going down
+    if self.speed_kph > LKAS_LIMITS.ENABLE_SPEED and self.low_speed_lockout:
+      self.low_speed_lockout = False
+    elif self.speed_kph < LKAS_LIMITS.DISABLE_SPEED and not self.low_speed_lockout:
+      self.low_speed_lockout = True
+
+    if (self.low_speed_lockout or self.steer_lkas.block) and self.speed_kph < LKAS_LIMITS.DISABLE_SPEED:
```

Comment:
self.steer_lkas block can just be local to the function. don't see it used anywhere else